src/compile/scale.ts

```diff
@@ -504,3 +506,78 @@ export function zero(scale: Scale, channel: Channel, fieldDef: FieldDef) {
   }
   return undefined;
 }
+
+/**
+ * Merge scales from children by unioning their domains
+ */
+export function mergeScales(model: LayerModel, channel: string) {
```

Comment:
it's so hard to diff across files.  In the future, can you avoid moving some part of the code across files in a big PR like this?  We can always add `//TODO: move to XXX.ts` and handle it later.